### ISSUE TITLE
npm WARN deprecated debug@3.2.6

### ISSUE THREAD

**Issue by rodrigolzd** [Ylianst/MeshCentral#3185]
Not sure what to do. Installing MeshCentral on a new, fully updated ubuntu 18.04 server gererates this warning:

```
meshcentral@server:~$ npm install meshcentral
npm WARN deprecated debug@3.2.6: Debug versions >=3.2.0 <3.2.7 || >=4 <4.3.1 have a low-severity ReDos regression when used in a Node.js environment. It is recommended you upgrade to 3.2.7 or 4.3.1. (https://github.com/visionmedia/debug/issues/797)

added 118 packages, and audited 119 packages in 39s

6 packages are looking for funding
  run `npm fund` for details

4 high severity vulnerabilities

Some issues need review, and may require choosing
a different dependency.

Run `npm audit` for details.
meshcentral@server:~$ npm i debug@latest

added 10 packages, changed 2 packages, and audited 129 packages in 2s

6 packages are looking for funding
  run `npm fund` for details

4 high severity vulnerabilities

Some issues need review, and may require choosing
a different dependency.

Run `npm audit` for details.
meshcentral@server:~$ npm audit
# npm audit report

nedb  *
Severity: high
Prototype Pollution - https://github.com/advisories/GHSA-339j-hqgx-qrrx
Depends on vulnerable versions of binary-search-tree
Depends on vulnerable versions of underscore
No fix available
node_modules/nedb
  meshcentral  *
  Depends on vulnerable versions of nedb
  node_modules/meshcentral

underscore  1.3.2 - 1.12.0
Severity: high
Arbitrary Code Execution in underscore - https://github.com/advisories/GHSA-cf4h-3jhx-xvhq
No fix available
node_modules/underscore
  binary-search-tree  *
  Depends on vulnerable versions of underscore
  node_modules/binary-search-tree
    nedb  *
    Depends on vulnerable versions of binary-search-tree
    Depends on vulnerable versions of underscore
    node_modules/nedb
      meshcentral  *
      Depends on vulnerable versions of nedb
      node_modules/meshcentral

4 high severity vulnerabilities

Some issues need review, and may require choosing
a different dependency.
meshcentral@server:~$

```

**Comment by Ylianst** [comment]
I need to look into this. NeDB is not being updated anymore and so, it's creating issues.

**Comment by Ylianst** [comment]
What version of NodeJS are you running?

**Comment by rodrigolzd** [comment]
As a matter of fact node was update today before I tried again and everything else is up to date. should I just ignore this error? Im just migration a server from one host to another in a different hosting company and country. I use mongodb, not sure what this error will effect me now. Im reporting only for debugging purposes.

```
meshcentral@mesh:/opt/meshcentral$ node -v
v16.11.1
```

and still gets me this message:

```
meshcentral@mesh:/opt/meshcentral$ npm install  meshcentral
npm WARN deprecated debug@3.2.6: Debug versions >=3.2.0 <3.2.7 || >=4 <4.3.1 have a low-severity ReDos regression when used in a Node.js environment. It is recommended you upgrade to 3.2.7 or 4.3.1. (https://github.com/visionmedia/debug/issues/797)

added 118 packages, and audited 119 packages in 26s

6 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities

```

**Comment by Ylianst** [comment]
Ok, I will try to replicate this.

**Comment by nzalev** [comment]
This is caused by a module, debug, used as a dependency for a number of other components. Version 3.2.6 is specified as an exact match by cookie-session from expressjs.

```json
  "repository": "expressjs/cookie-session",
  "dependencies": {
    "cookies": "0.8.0",
    "debug": "3.2.6",
    "on-headers": "~1.0.2",
    "safe-buffer": "5.2.0"
  },
```

Other packages, such as handlebars have dependencies with `^3.2.6`. I don't know what that would resolve to if not for cookie-seesion.

**Comment by Ylianst** [comment]
Fixed, will be in MeshCentral 0.9.34. Let me know if it works. I change the cookie-session dependency to v1.4.0 which seems to be ok.
